torch/distributed/__init__.py

```diff
@@ -85,7 +80,11 @@ def init_process_group(backend, init_method='env://', **kwargs):
     elif backend == "nccl":
         _backend = dist_backend.NCCL
     else:
-        raise RuntimeError("Invalid distributed backend name detected")
+        raise RuntimeError("Invalid distributed backend name: " + backend)
```

Comment:
@apaszke  Did this change so that the wrong backend string won't be passed into the torch._C function. We will get an error like this.

```
tengli@learnfairXXXX:~$ python -u imagenet/main.py -j 16 -a resnet50 --print-freq 1 --batch-size 256 --world-size 2 --dist-url 'tcp://XXX.XXX.XXX.XXX:20000' /datasets/XXXX/XXXX/ --epochs 10 --dist-backend 'abc' --rank 0
Traceback (most recent call last):
  File "/private/home/tengli/pytorch_examples/imagenet/main_distributed.py", line 371, in <module>
    main()
  File "/private/home/tengli/pytorch_examples/imagenet/main_distributed.py", line 78, in main
    rank=args.rank)
  File "/private/home/tengli/miniconda3/lib/python3.6/site-packages/torch/distributed/__init__.py", line 83, in init_process_group
    raise RuntimeError("Invalid distributed backend name: " + backend)
RuntimeError: Invalid distributed backend name: abc
```